frontend/src/views/RunView.vue

```diff
@@ -32,6 +35,37 @@ let activeWell = reactive({
 let user = ref(null);
 getUserName((email) => { user.value = email }).then();
 
+watch(() => route.query, (after, before) => {
+  console.log(before)
+  if (
+    (after.label || after.run)
+    && (
+      before == undefined
+      || (
+        before.label && before.run && after.label != before.label && after.run != before.run
```

Comment:
It is conceavable for teh value of run not to change. Therefore, the condition should be
`&& (after.label != before.label || after.run != before.run)`